### PR TITLE
feat: set err.headerSent before app error event emit

### PR DESCRIPTION
**Pull request by fengmk2**

### CODE REVIEW

File: test/context/onerror.js [hunk 1]
```diff
@@ -86,6 +86,29 @@ describe('ctx.onerror(err)', () => {
     });
   });
 
+  it('should ignore error after headerSent', done => {
+    const app = new Koa();
+
+    app.on('error', err => {
+      err.message.should.equal('mock error');
+      err.headerSent.should.equal(true);
```
Comment (dead-horse) [inline test/context/onerror.js]: the only difference is we can get `err.headerSent` in app's error listener?

Comment (jonathanong) [inline test/context/onerror.js]: I think so

**Comment by jonathanong** [comment]
More of a feature than a fix but 👍

**Comment by dead-horse** [comment]
@fengmk2 make a PR to v1.x please.

File: test/context/onerror.js (hunk 1 above)
Comment (fengmk2) [inline test/context/onerror.js]: yes

**Comment by fengmk2** [comment]
@dead-horse ok

**Pull request merged**